python/IAM_POLICY_REQUIRED/IAM_POLICY_REQUIRED.py

```diff
@@ -0,0 +1,405 @@
+            policy['PolicyName'] for policy in client.list_attached_group_policies(GroupName=group)['AttachedPolicies']
+        ])
+
+    return policies
+
+
+def get_role_policies(role_name):
+    client = boto3.client('iam')
+    # Get the users managed policies
+    return [
+        policy['PolicyName'] for policy in client.list_attached_role_policies(RoleName=role_name)['AttachedPolicies']
+    ]
+
+
+def has_policy_attached(resource_name, resource_type, policy_name):
+    if resource_type == 'AWS::IAM::User':
+        return policy_name in get_user_policies(resource_name)
+    elif resource_type == 'AWS::IAM::Role':
+        return policy_name in get_role_policies(resource_name)
+    else:
+        raise ValueError('Unable to handle resource type {}'.format(resource_type))
+
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    """Form the evaluation(s) to be return to Config Rules
+
+    Return either:
+    None -- when no result needs to be displayed
+    a string -- either COMPLIANT, NON_COMPLIANT or NOT_APPLICABLE
+    a dictionary -- the evaluation dictionary, usually built by build_evaluation_from_config_item()
+    a list of dictionary -- a list of evaluation dictionary , usually built by build_evaluation()
+
+    Keyword arguments:
+    event -- the event variable given in the lambda handler
+    configuration_item -- the configurationItem dictionary in the invokingEvent
+    valid_rule_parameters -- the output of the evaluate_parameters() representing validated parameters of the Config Rule
+
+    Advanced Notes:
+    1 -- if a resource is deleted and generate a configuration change with ResourceDeleted status, the Boilerplate code will put a NOT_APPLICABLE on this resource automatically.
+    2 -- if a None or a list of dictionary is returned, the old evaluation(s) which are not returned in the new evaluation list are returned as NOT_APPLICABLE by the Boilerplate code
+    3 -- if None or an empty string, list or dict is returned, the Boilerplate code will put a "shadow" evaluation to feedback that the evaluation took place properly
+    """
+    ignored_roles = valid_rule_parameters["ignored_roles"]
+    ignored_users = valid_rule_parameters["ignored_users"]
+    policy_name = valid_rule_parameters['policy_name']
+
+    if should_ignore_config_item(configuration_item, ignored_roles, ignored_users):
+        return 'NOT_APPLICABLE'
+    elif has_policy_attached(configuration_item['resourceName'], configuration_item['resourceType'], policy_name):
+        return 'COMPLIANT'
+    else:
+        return 'NON_COMPLIANT'
+
+
+def evaluate_parameters(rule_parameters):
+    """Evaluate the rule parameters dictionary validity. Raise a ValueError for invalid parameters.
+
+    Return:
+    anything suitable for the evaluate_compliance()
+
+    Keyword arguments:
+    rule_parameters -- the Key/Value dictionary of the Config Rules parameters
+    """
+    valid_rule_parameters = rule_parameters

```

Comment:
I'd suggest to use the ExceptionList the same way the service team is doing: https://docs.aws.amazon.com/config/latest/developerguide/iam-policy-blacklisted-check.html

Reply:
good call - I wonder if we should use a list of policyArns instead of just 1 policy for consistency with the IAM Policy Blacklisted Check?

Reply:
+1

Reply:
is there an example of dealing with comma separated lists (does it come in as a list or a string I have to split) and parameters structured like the ExceptionList for the example you gave i.e. `users:[user1, user2],roles:[role1, role2]`? Just want to make sure i'm not reinventing the wheel for parsing the args

Reply:
fixed